sheepdog/transactions/upload/sub_entities.py

```diff
@@ -187,6 +187,9 @@ def flush_to_session(self):
         if not self.node:
             return
 
+        if self.transaction.dry_run:
```

Comment:
This bypasses the `super(FileUploadEntity, self).flush_to_session()` we still need to run for dry run